### ISSUE TITLE
IgnorePkg broken

### ISSUE THREAD

**Issue by kattjevfel** [Morganamilo/paru#126]
This appears to have been broken for a while, but it hasn't annoyed me enough until now.

I've set `IgnorePkg   = dolphin-emu-git pipewire-git`, but yet all I get is this:
```
katt@main ~ » paru -Syu                                                                                                                                                                                                        130 ↵
no new news
:: Synchronising package databases...
 kde-unstable.db is up to date
 testing.db is up to date
 core.db is up to date
 extra.db is up to date
 community-testing.db is up to date
 community.db is up to date
 multilib-testing.db is up to date
 multilib.db is up to date
:: Starting full system upgrade...
 there is nothing to do
:: Looking for AUR upgrades
:: Looking for devel upgrades
warning: pipewire-git: ignoring package upgrade (0.3.17.r35.g3c2ab98a-1 => 0.3.17.35.g3c2ab98a-1)
:: Resolving dependencies...
:: Calculating conflicts...
:: Calculating inner conflicts...
:: Packages not in the AUR: gst-plugin-pipewire-git

Repo Make (29) ninja-1.10.2-1  meson-0.56.0-3  doxygen-1.8.20-1  gts-0.7.6.121130-2  graphviz-2.44.1-6  perl-encode-locale-1.05-7  perl-http-date-6.05-3  perl-file-listing-6.11-1  perl-html-tagset-3.20-10  
    perl-clone-0.45-2  perl-io-html-1.004-1  perl-lwp-mediatypes-6.02-8  perl-uri-5.05-1  perl-http-message-6.26-1  perl-html-parser-3.75-1  perl-http-cookies-6.09-1  perl-http-daemon-6.06-2  perl-http-negotiate-6.01-8  
    perl-net-http-6.19-4  perl-try-tiny-0.30-5  perl-www-robotrules-6.02-8  perl-libwww-6.49-1  perl-xml-parser-2.46-2  xmltoman-0.4-4  valgrind-3.16.1-1  vulkan-headers-1:1.2.160-1  libuv-1.40.0-1  cmake-3.19.1-2  
    python-setuptools-scm-4.1.2-3
Aur (5) pipewire-git-0.3.17.35.g3c2ab98a-1 (pipewire-alsa-git pipewire-docs-git pipewire-jack-git pipewire-pulse-git)  ruffle-nightly-bin-2020.12.03-1  dolphin-emu-git-5.0.r11879.gc9aab4f809-1  
    gallery-dl-git-1.14.2.r14.g27d163af-1  protontricks-git-1.4-1

:: Proceed to review? [Y/n]:
```

Strangely only pipewire-git is even listed as being ignored (though that gets thrown out the window a few lines later anyway), even though dolphin-emu-git should be ignored.

dolphin-emu-git is in devel.json but pipewire-git isn't (it's completely handled outside of paru as I use my own pkgbuild), so it's not isolated to it being "devel" package or so.

Adding the packages to "NoWarn" (which I still don't get the point of) didn't help either.

`paru v1.1.3.r3.g70af4ab +git - libalpm v12.0.1`

**Comment by Morganamilo** [comment]
> Strangely only pipewire-git is even listed as being ignored (though that gets thrown out the window a few lines later anyway)

`pipewire-git-0.3.17.35.g3c2ab98a-1 (pipewire-alsa-git pipewire-docs-git pipewire-jack-git pipewire-pulse-git)`

pipewire-git is the pkgbase, the packages being upgraded are `pipewire-alsa-git pipewire-docs-git pipewire-jack-git pipewire-pulse-git` perhaps you want to ignore them too?.

> Adding the packages to "NoWarn"

NoWarn is generally for your local pkgbuilds where you don't want paru to complain it's not in the aur.

**Comment by kattjevfel** [comment]
I realised about the pkgbase thing, my bad. Still, it doesn't ignore the other one, which is triggered by being a "devel" package.

**Comment by Morganamilo** [comment]
Ah, that should be handled but I'll take a look.

**Comment by nichobi** [comment]
```
nichobi@soyuz:~$ paru
:: Synchronising package databases...
 core is up to date
 extra is up to date
 community is up to date
 multilib is up to date
 nichobi is up to date
:: Starting full system upgrade...
 there is nothing to do
:: Looking for AUR upgrades
:: Looking for devel upgrades
warning: htop-vim-git: ignoring package upgrade (1539.0806a79-1 => 1697.d1db9da-1)
1 devel/htop-vim-git  1539.0806a79-1 -> latest-commit
:: Packages to exclude:
```
I'm also seeing this issue. Despite it saying the package is ignored, it is then immediately included for update.

**Comment by Morganamilo** [comment]
I believe this was working. Looks like I may have forgotten to handle ignore pkg when refactoring the devel upgrading.

**Comment by Morganamilo** [comment]
This was fixed with ac0de9f, Just pointed it to the wrong issue.
